### PR TITLE
Store and load the planner related depth info of a cylinder in logfile

### PR DESCRIPTION
**Pull request by sfuchs79**
### Describe the pull request:

- [ ] Bug fix
- [ ] Functional change
- [x] New feature
- [ ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

Store cylinder.depth in XML files and in git storage.
This info is in fact the gas switch depth of a specific gas/cylinder
in the planner.
This change avoids the need of typing in a user specific depth value
again when replanning an existing planned dive.

### Mentions:

@atdotde @RickMWalsh @janmulder

### CODE REVIEW

File: core/load-git.c [hunk 1]
```diff
@@ -390,6 +390,10 @@ static void parse_cylinder_keyvalue(void *_cylinder, const char *key, const char
 		cylinder->cylinder_use = cylinderuse_from_text(value);
 		return;
 	}
+	if (!strcmp(key, "depth")) {
```
Comment (neolit123) [inline core/load-git.c:393]: i think we should use the `same_string` function here. it is safer.

**Review by neolit123 — CHANGES_REQUESTED** [review]
thanks @sfuchs79. only one small change request.
other than that lgtm.

@mturkia please, have a look if you have 5 minutes. thanks.

**Comment by mturkia** [comment]
LGTM, but I wonder what the CI error is. Doesn't really look like this patch should affect this..

```
[ 86%] Generating qrc_kirigami.cpp

make[2]: *** No rule to make target `../mobile-widgets/qml/kirigami/icons/go-next.svg', needed by `qrc_mobile-resources.cpp'.  Stop.

make[2]: *** Waiting for unfinished jobs....

[ 87%] Generating qrc_subsurface.cpp

make[1]: *** [CMakeFiles/subsurface-mobile_autogen.dir/all] Error 2

make[1]: *** Waiting for unfinished jobs....
```

**Comment by dirkhh** [comment]
From the build log:
```
Cloning into 'breeze-icons'...
fatal: unable to connect to anongit.kde.org:
anongit.kde.org[0: 159.203.184.29]: errno=Connection timed out
anongit.kde.org[1: 2604:a880:400:d0::666:c001]: errno=Cannot assign requested address
./scripts/mobilecomponents.sh: line 43: pushd: breeze-icons: No such file or directory
fatal: Not a git repository (or any of the parent directories): .git
./scripts/mobilecomponents.sh: line 45: popd: directory stack empty
cp: cannot stat ‘breeze-icons/icons/actions/24/dialog-cancel.svg’: No such file or directory
```
Looks like a temporary failure cloning the icons...

**Comment by sfuchs79** [comment]
CI error:
I think we had s.th. like this just recently -> network issue when downloading s.th.?!

same_string:
As usually my code is only copy&paste from old code ;-)
What I want to say: If we want to change to same_string then every strcmp in this file and that will be a separate commit. Up to you if I should do it. But risk that this conflicts with @mturkia s open PR.

Note:
@dirkhh triggered Linus to also look at this, so let's wait for Linus... ah looking at my mail Linus already said ok while I was typing - great.

**Comment by dirkhh** [comment]
I restarted the build job for Linux

**Comment by dirkhh** [comment]
so same_string() is just a string compare that is smart about handling null ptrs. Which means that if you KNOW that neither pointer is null (one of them is a constant and therefore never null) then you might as well use strcmp.
Looking through the code I don't think key could ever be null...

**Review by neolit123 — APPROVED** [review]

**Comment by atdotde** [comment]
Looks good.

**Pull request merged**